[quantization] Factored out the default mappings

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27183 [quantization] Replacing the skip_list with white_list in the qconfig propagation
* **#27164 [quantization] Factored out the default mappings**

Differential Revision: [D17694475](https://our.internmc.facebook.com/intern/diff/D17694475)

Commit:
[quantization] Factored out the default mappings

Commit:
Update on "[quantization] Factored out the default mappings"

jerryzh168 commented on torch/quantization/quantize.py:
```diff
@@ -85,10 +62,10 @@ def propagate_qconfig_(module, qconfig_dict=None):
-            configuration, qconfig applies to all submodules of a given
-            module unless qconfig for the submodules are specified (when the
-            submodule already has qconfig attribute)
+        qconfig_dict: dictionary that maps from name or type of submodule to
```
btw, do you have a python formatting tool

zafartahirov replied:
No -- doing it manually LOL. It was just irritating to see it go off screen :)

jerryzh168 reviewed (approved):
LG Thanks!

jerryzh168 commented on torch/quantization/__init__.py:
```diff
@@ -14,7 +15,7 @@ def default_eval_fn(model, calib_data):
 
 _all__ = [
-    'QuantWrapper', 'QuantStub', 'DeQuantStub', 'DEFAULT_MODULE_MAPPING',
+    'QuantWrapper', 'QuantStub', 'DeQuantStub',
```
this mapping is exposed for users who want to extend the mapping with their own entries. I think it might be useful to expose them?

raghuramank100 replied:
DEFAULT_MODULE_MAPPING is an argument to the convert/prepare_qat/quantize and since we allow for users to change it, not sure if this is needed.

zafartahirov replied:
So, I am landing it as is in this PR?

Commit:
Update on "[quantization] Factored out the default mappings"


Differential Revision: [D17694475](https://our.internmc.facebook.com/intern/diff/D17694475)

Commit:
Update on "[quantization] Factored out the default mappings"


Differential Revision: [D17694475](https://our.internmc.facebook.com/intern/diff/D17694475)

Commit:
Update on "[quantization] Factored out the default mappings"


Differential Revision: [D17694475](https://our.internmc.facebook.com/intern/diff/D17694475)

Commit:
Update on "[quantization] Factored out the default mappings"


Differential Revision: [D17694475](https://our.internmc.facebook.com/intern/diff/D17694475)

zafartahirov commented:
Replacing the default arguments with `None` => This makes sure that the default args are immutable

Commit:
Update on "[quantization] Factored out the default mappings"


Differential Revision: [D17694475](https://our.internmc.facebook.com/intern/diff/D17694475)

Commit:
Update on "[quantization] Factored out the default mappings"


Differential Revision: [D17694475](https://our.internmc.facebook.com/intern/diff/D17694475)
